src/kafka/decam.rs

```diff
@@ -0,0 +1,247 @@
+use crate::{
+    conf,
+    kafka::base::{consume_partitions, AlertConsumer, AlertProducer, ConsumerError},
+    utils::{
+        data::count_files_in_dir,
+        o11y::{as_error, log_error},
+    },
+};
+use indicatif::ProgressBar;
+use rdkafka::config::ClientConfig;
+use rdkafka::producer::{FutureProducer, FutureRecord, Producer};
+use redis::AsyncCommands;
+use tracing::{error, info, instrument};
+
+const DECAM_SERVER_URL: &str = "localhost:9092";
+
+pub struct DecamAlertConsumer {
+    output_queue: String,
+    n_threads: usize,
+    max_in_queue: usize,
+    group_id: String,
+    server: String,
+    config_path: String,
+}
+
+impl DecamAlertConsumer {
+    pub fn new(
+        n_threads: usize,
+        max_in_queue: Option<usize>,
+        output_queue: Option<&str>,
+        group_id: Option<&str>,
+        server: Option<&str>,
+        config_path: &str,
+    ) -> Self {
+        if 15 % n_threads != 0 {
+            panic!("Number of threads should be a factor of 15");
+        }
```

Comment:
The other surveys take a different approach to this as of #201. Do you want to make those changes here, too?

Reply:
Yes, this PR pre dates 201 so the changes that are in there aren't in sync yet.